libraries/lib-project-file-io/ProjectFileIO.cpp

```diff
@@ -785,6 +785,20 @@ bool ProjectFileIO::CheckVersion()
       );
       return false;
    }
+   else if (version < BaseProjectFormatVersion) {
+      using namespace BasicUI;
+      wxString currentVersionStr = wxString::Format("%u.%u", BaseProjectFormatVersion.Major, BaseProjectFormatVersion.Minor);
+      bool updateVersion = (MessageBoxResult::Yes == ShowMessageBox(
+         XO("This project was created using an older Audacity version. "
+            "Once saved, the project can only be opened with Audacity version %s or newer.\n\n"
+            "Do you want to continue? This action cannot be undone.").Format(currentVersionStr),
+         MessageBoxOptions{}
+            .Caption(XO("Update Audacity project version?"))
+            .ButtonStyle(Button::YesNo)));
+      if ( !updateVersion ) {
+         return false;
+      }
```

Comment:
i think we should skip this bit: 

* Yes/no has the issue of always requiring quite a bit of brain energy to parse. 
* I don't think "this action cannot be undone" is true - if you close the project without saving, you won't be updating the project version, will you?

I think we can just keep the upper part with an OK dialog box.

Reply:
Updated